lucidity/rpc/rpc.go

```diff
@@ -246,3 +205,45 @@ func (s *server) IsValidVersion(v string) bool {
 	_, valid := s.validVersions.Load(v)
 	return valid
 }
+
+// Describe sends prometheus description of all metrics collected in Collect.
+func (s *server) Describe(out chan<- *prometheus.Desc) {
+	liveWorkers.Describe(out)
+	deadWorkers.Describe(out)
+	healthyWorkers.Describe(out)
+	unhealthyWorkers.Describe(out)
+	busyWorkers.Describe(out)
+}
+
+// Collect counts number of unhealthy, dead and busy workers and send the
+// corresponding metrics. It also checks the worker health on the fly, in case
+// the version or the updateTime is no longer valid.
+func (s *server) Collect(out chan<- prometheus.Metric) {
+	var total, unhealthy, dead, busy float64
+	s.workers.Range(func(_, v interface{}) bool {
+		r := v.(*pb.UpdateRequest)
+		s.checkWorkerHealth(r)
+		if !r.Healthy {
```

Comment:
nit: might be nicer to use a switch statement here

Reply:
Oh actually, we dont want a switch here. We want to go through all `if` blocks.